### ISSUE TITLE
Executable propagator fails to set loader for dataset with already initialized config

### ISSUE THREAD

**Issue by synapticarbors** [westpa/westpa#115]
There is a problematic bit of code in the `ExecutablePropagator` that can mishandle setting dataset loaders during initialization when the `rc` has already had it's loaders set:

https://github.com/westpa/westpa/blob/master/src/west/propagators/executable.py#L165-L171

```python
loader_directive = dsinfo.get('loader')
if loader_directive:
    loader = get_object(loader_directive)
elif dsname != 'pcoord':
    loader = aux_data_loader

dsinfo['loader'] = loader
```

Here, `loader_directive` is assumed to be a `str` or `None`. If `dsinfo['loader'] is not None`, then it gets passed to `get_object` that assumes its argument is a `str` and converts this string into a function that then overwrites `dsinfo['loader']`. 

The problem is that if a function has already replaced `dsinfo['loader']` it will evaluate `if loader_directive` also to `True` and get passed to `get_object`, which will now fail with:

```python
(modspec, symbol) = object_name.rsplit('.', 1)

AttributeError: 'function' object has no attribute 'rsplit'
```

@atbogetti discovered this in the westpa-2.0 branch in the `w_init` when I had inadvertently deleted what looked like an unused variable, but had the side-effect of loading this before calling out to the work_manager. When using the threads work manager, the first call to setup the propagator worked, but since it was using a shared `rc` among the threads, subsequent calls failed since the loader was set as a function.

The proper way to handle this would probably be something like:

```python
loader_directive = dsinfo.get('loader')
if callable(loader_directive):
    loader = loader_directive
elif isinstance(loader_directive, str):
    loader = get_object(loader_directive)
elif dsname != 'pcoord':
    loader = aux_data_loader

dsinfo['loader'] = loader
```

**Comment by dnlebard** [comment]
@synapticarbors @atbogetti First, I agree with Joshua's fix.

Second, a complete nitpick and aside, but can we get away from using hardcoded strings like ```'pcoord'``` and ```'loader'```, and instead make a set of common variables that can be imported and contain those strings? I find it easy to make to make simple string typing mistakes (even with an editor that has autocompletion), and variables make these mistakes less likely for future/new developers. I would be happy to raise an issue for this type of common-string refactoring if you think it's worth it.

**Comment by synapticarbors** [comment]
@dnlebard Thanks for the feedback. I think the idea of getting away from hardcoded strings and instead using either variable names or enums as appropriate, is generally a good idea. I don't have a design in mind, but if you want to raise a separate issue so that we can discuss it further, that would be great.

**Comment by jeremyleung521** [comment]
Fixed by #293.
